source/common/http/http1/balsa_parser.cc

```diff
@@ -163,8 +163,19 @@ void BalsaParser::OnBodyChunkInput(absl::string_view input) {
   connection_->bufferBody(input.data(), input.size());
 }
 
-void BalsaParser::OnHeaderInput(absl::string_view /*input*/) {}
-void BalsaParser::OnTrailerInput(absl::string_view /*input*/) {}
+void BalsaParser::OnHeaderInput(absl::string_view input) {
+  // NUL character is not allowed anywhere in the start line or in the header section.
+  if (input.find(absl::string_view("\0", 1)) != absl::string_view::npos) {
```

Comment:
very optional, here and below: Can use absl::StrContains()?

Reply:
Good idea, done.  Incidentally I used a suboptimal signature, searching for a substring of length 1 instead of searching for a char.  (Both absl::string_view::find() and absl::StrContains() support both call signatures.)